### ISSUE TITLE
Only java supported?

### ISSUE THREAD

**Issue by tg44** [Chentai-Kao/call-graph-plugin#20]
At [this line](https://github.com/Chentai-Kao/call-graph-plugin/blob/2aac0f24cf87efcd02012ca194ee7b88b99ad7f6/src/main/kotlin/callgraph/Utils.kt#L94) can't we use [PsiClassOwner](https://upsource.jetbrains.com/idea-ce/file/idea-ce-738b4c9365fbc976485aa7f152053a70ccac42ae/java/java-psi-api/src/com/intellij/psi/PsiClassOwner.java) instead of PsiJavaFile?

I assume with this "minimal" mod, the plugin would work with a lot of languages (C#, Scala, Kotlin).

I will try this out!

**Comment by 5ay3h** [comment]
@tg44 any success?

**Comment by tg44** [comment]
Nope, this repo feeled like a big rabbithole to me :D I like the idea, but I can't get a gasp of the codebase... BTW in theory we could use intellijs build in language processor backed [usages API](https://plugins.jetbrains.com/docs/intellij/find-usages.html). That was what I get in 1 hour, and I left it bcs I had other things to do... (Simply changing the lines that I mentioned is not enough. The whole problem is far more complex.)
